### PR TITLE
PSW: fix docker build and psw bin installer for urts version update

### PR DESCRIPTION
**Pull request by haitaohuang**
Signed-off-by: Haitao Huang <haitao.huang@linux.intel.com>

### CODE REVIEW

File: linux/installer/common/psw/Makefile [hunk 1]
```diff
@@ -39,13 +38,14 @@ SOURCE_INSTALL_PATH=$(INSTALL_PATH)
...
 PSW_LIB_PATH=$(SOURCE_INSTALL_PATH)/$(LIB_DIR)
 USR_LIB_PATH=$(addprefix $(DESTDIR)/usr/, $(or $(notdir $(shell gcc -print-multi-os-directory 2> /dev/null)),$(if $(wildcard /usr/lib64/.*),lib64),lib))
-
 SPLIT_VERSION=$(word $2,$(subst ., ,$1))
 
 ECL_VER=1.0.0
 LCH_VER=1.0.0
 EPID_VER=1.0.0
 QEX_VER=1.0.0
+URTS_FULL_VER:= 2.0.100.3
```
Comment (llly) [inline linux/installer/common/psw/Makefile:47]: `URTS_VER:= 2.0.0` is enough for bin installer. This version is not updated until major version changes.

File: linux/installer/common/psw/Makefile [hunk 2]
```diff
@@ -63,6 +63,9 @@ install:
 	mv $(PSW_LIB_PATH)/libsgx_enclave_common.so $(USR_LIB_PATH)
 	rmdir $(PSW_LIB_PATH)
 	cd $(USR_LIB_PATH) && \
+	mv libsgx_urts.so libsgx_urts.so.$(URTS_FULL_VER) && \
+	ln -fs libsgx_urts.so.$(URTS_FULL_VER) libsgx_urts.so.$(URTS_MAJOR_VER) && \
```
Comment (llly) [inline linux/installer/common/psw/Makefile:67]: Bbetter to remove `URTS_MAJOR_VER` and call `SPLIT_VERSION` as the other versions.

File: docker/build/Dockerfile [hunk 3]
```diff
@@ -58,7 +58,7 @@ RUN make sdk_install_pkg_no_mitigation
 
 WORKDIR /opt/intel
 RUN sh -c 'echo yes | /linux-sgx/linux/installer/bin/sgx_linux_x64_sdk_*.bin'
-
```
Comment (llly) [inline docker/build/Dockerfile:61]: No need to remove empty line. 
Same comments to the other empty lines change.

**Review by xxu36 — APPROVED** [review]

File: linux/installer/common/psw/BOMs/psw_base.txt [hunk 4]
```diff
@@ -9,8 +9,8 @@ DeliveryName	InstallName	FileCheckSum	FileFeature	FileOwner
 <deliverydir>/build/linux/libsgx_default_qcnl_wrapper.so	<installdir>/package/aesm/libsgx_default_qcnl_wrapper.so	0	main	STP
 <deliverydir>/build/linux/liburts_internal.so	<installdir>/package/aesm/liburts_internal.so	0	main	STP
 <deliverydir>/build/linux/libCppMicroServices.so.4.0.0	<installdir>/package/aesm/libCppMicroServices.so.4.0.0	0	main	STP
-<deliverydir>/external/dcap_source/QuoteGeneration/build/linux/libsgx_pce_logic.so	<installdir>/package/aesm/libsgx_pce_logic.so.1	0	main	STP
-<deliverydir>/external/dcap_source/QuoteGeneration/build/linux/libsgx_qe3_logic.so	<installdir>/package/aesm/libsgx_qe3_logic.so.1	0	main	STP
+<deliverydir>/external/dcap_source/QuoteGeneration/build/linux/libsgx_pce_logic.so	<installdir>/package/aesm/libsgx_pce_logic.so.1	main	STP
+<deliverydir>/external/dcap_source/QuoteGeneration/build/linux/libsgx_qe3_logic.so	<installdir>/package/aesm/libsgx_qe3_logic.so	main	STP
```
[the remark this answers is not in the document]
Comment (haitaohuang) [reply linux/installer/common/psw/BOMs/psw_base.txt:13]: Thanks pce_logic change was made by accident. aesm will complaining missing qe3_logic.so if I don't make that change but not complaining for pce_logic. It seems inconsistent, any ideas?

[the remark this answers is not in the document]
Comment (llly) [reply linux/installer/common/psw/BOMs/psw_base.txt:13]: Add a link as the other libs in `Makefile`.

File: linux/installer/common/psw/Makefile [hunk 5]
```diff
@@ -63,6 +64,9 @@ install:
 	mv $(PSW_LIB_PATH)/libsgx_enclave_common.so $(USR_LIB_PATH)
 	rmdir $(PSW_LIB_PATH)
 	cd $(USR_LIB_PATH) && \
+	mv libsgx_urts.so libsgx_urts.so.$(URTS_VER) && \
```
[the remark this answers is not in the document]
Comment (haitaohuang) [reply linux/installer/common/psw/Makefile:67]: Is it really needed? quote and launch plugins seem to only link to urts_internal, no dynamic link to urts from AESM.
And other Makefiles do the same: 
linux/installer/common/sgx-aesm-service/Makefile
102:    ln -fs liburts_internal.so libsgx_urts.so
linux/installer/common/psw-dcap/Makefile
221:    ln -fs liburts_internal.so libsgx_urts.so

[the remark this answers is not in the document]
Comment (llly) [reply linux/installer/common/psw/Makefile:68]: Yes. `libsgx_pce_logic.so.1` requires this special urts to work in aesm_service and doesn't `dlopen` `libsgx_launch.so`. We also need to fix the other Makefiles.

**Review by llly — APPROVED** [review]

**Pull request merged**